internal/pkg/githubapi/github.go

```diff
@@ -104,6 +104,30 @@ func (ghPrClientDetails *GhPrClientDetails) getBlameURLPrefix() string {
 	return fmt.Sprintf("%s/%s/%s/blame", githubHost, ghPrClientDetails.Owner, ghPrClientDetails.Repo)
 }
 
+// This function is used to check if we should display the sync branch checkbox in the PR comment.
+// This depends on the allowSyncfromBranchPathRegex configuration and whether the component is new or not.
+func shouldSyncBranchCheckBoxBeDisplayed(componentPathList []string, allowSyncfromBranchPathRegex string, diffOfChangedComponents []argocd.DiffResult) bool {
+	var displaySyncBranchCheckBox bool
+
+out:
+	for _, componentPath := range componentPathList {
+		if isSyncFromBranchAllowedForThisPath(allowSyncfromBranchPathRegex, componentPath) {
```

Comment:
Base [The happy path is left-aligned.](https://maelvls.dev/go-happy-line-of-sight/) I would suggest:
``
if !condition {
   return false
}
```